catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -58,30 +60,23 @@ public String[] listPartitionNames() {
 
   @Override
   public Partition[] listPartitions() {
-    List<String> partitionNames;
     List<org.apache.hadoop.hive.metastore.api.Partition> partitions;
     try {
-      partitionNames =
-          table
-              .clientPool()
-              .run(c -> c.listPartitionNames(table.schemaName(), table.name(), (short) -1));
       partitions =
           table
               .clientPool()
-              .run(c -> c.getPartitionsByNames(table.schemaName(), table.name(), partitionNames));
+              .run(c -> c.listPartitions(table.schemaName(), table.name(), (short) -1));
     } catch (TException | InterruptedException e) {
       throw new RuntimeException(e);
     }
-
-    // should never happen
-    Preconditions.checkArgument(
-        partitionNames.size() == partitions.size(),
-        "oops?! partition names and partitions size are not equal: %s vs %s",
-        partitionNames.size(),
-        partitions.size());
-
-    return IntStream.range(0, partitionNames.size())
-        .mapToObj(i -> fromHivePartition(partitionNames.get(i), partitions.get(i)))
+    List<String> partCols =
+        table.buildPartitionKeys().stream().map(FieldSchema::getName).collect(Collectors.toList());
+
+    return partitions.stream()
+        .map(
+            partition ->
+                fromHivePartition(
+                    FileUtils.makePartName(partCols, partition.getValues()), partition))
```

Comment:
Is it necessary to ensure that `partCols` and `partition.getValues()` have corresponding orders?

Reply:
Already orderly.

https://github.com/apache/hive/blob/0f8190fefa56513a042566ef87b852e8343610aa/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java#L5243-L5255